### PR TITLE
fix(maven-workspace): SNAPSHOT versions should not be included in the manifest

### LINKED ISSUE #1718: Release notes duplication in new PR design.
Release:please seems to be populating old release notes in the new release PR, though the versions being suggested are new. 
Reference: 
1. Latest release:main [PR](https://github.com/googleapis/google-cloud-java/pull/8655) (to be released)
2. Previous release:main [PR](https://github.com/googleapis/google-cloud-java/pull/8623) (already released)

### PR DESCRIPTION
**Pull request by chingor13**
Fixes #1718

Example new PR: https://github.com/googleapis/google-cloud-java/pull/8699

### CODE REVIEW

File: src/plugins/maven-workspace.ts [hunk 1]
```diff
@@ -310,6 +315,16 @@ export class MavenWorkspace extends WorkspacePlugin<MavenArtifact> {
...
   }
 
+  /**
+   * Given a release version, determine if we should bump the manifest
+   * version as well. For maven artifacts, SNAPSHOT versions are not
+   * considered releases.
+   * @param {Version} version The release version
+   */
+  protected isReleaseVersion(version: Version): boolean {
+    return !version.preRelease?.includes('SNAPSHOT');
```
Comment (bcoe) [inline src/plugins/maven-workspace.ts:325]: nit: might be worth pulling `SNAPSHOT` into a constant, if we use this value in a variety of places.

**Review by bcoe — APPROVED** [review]
LGTM, if I'm understanding correctly, prior to this change [this file](https://github.com/googleapis/google-cloud-node/blob/main/.release-please-manifest.json#L9) would have been updated to include `-SNAPSHOT` versions, prior to this fix.

**Comment by chingor13** [comment]
cc @suztomo

**Comment by chingor13** [comment]
> if I'm understanding correctly, prior to this change [this file](https://github.com/googleapis/google-cloud-node/blob/main/.release-please-manifest.json#L9) would have been updated to include `-SNAPSHOT` versions, prior to this fix

Correct, the core manifest already knows not to update non-release versions into the manifest, but the workspace plugin did not.

**Pull request merged**